packages/core/src/compiler.ts

```diff
@@ -23,31 +29,29 @@ import {
   Saved,
   Unsaved,
 } from './interfaces';
-import { ensureTrailingSlash, getBasenameAndExtension } from './utils';
+import { cardURL, ensureTrailingSlash, getBasenameAndExtension } from './utils';
 import { getFileType } from './utils/content';
 import { CardstackError, BadRequest, augmentBadRequest, isCardstackError } from './utils/errors';
 import { hashCardFields } from './utils/fields';
-import babelPluginCardSerializerAnalyze from './babel-plugin-card-serializer-analyze';
 
-export const baseCardURL = 'https://cardstack.com/base/base';
-
-function getNonAssetFilePaths(sourceCard: RawCard<Unsaved>): (string | undefined)[] {
-  let paths: (string | undefined)[] = [];
-  for (const feature of FEATURE_NAMES) {
-    paths.push(sourceCard[feature]);
-  }
-  return paths.filter(Boolean);
-}
+const BASE_CARD_ID: CardId = {
+  realm: 'https://cardstack.com/base/',
+  id: 'base',
+};
+export const BASE_CARD_URL = cardURL(BASE_CARD_ID);
 
 export class Compiler<Identity extends Saved | Unsaved = Saved> {
   private builder: TrackedBuilder;
   private cardSource: RawCard<Identity>;
-  private modules: CompiledCard<Unsaved, LocalRef>['modules'];
+
+  private sourceModules: CompiledCard<Unsaved, LocalRef>['modules'];
+  private globalModules: CompiledCard<Unsaved, LocalRef>['modules'];
```

Comment:
1. I find it surprising that sourceModules and globalModules should really have the same type. 
2. I don't think "global" is a good name here. The distinction is more input vs output. Even the output are still local refs, not global refs.

Reply:
@ef4 Yeah. I wondered about it but didn't really push the thought as far as I could. I'll ponder.

Reply:
@ef4 I just did another pass. It's a subtle split up (at least for now, until we consolidate the rest of the analyze work)